src/lib.rs

```diff
@@ -1136,14 +1147,73 @@ impl<T> Drop for Async<T> {
     }
 }
 
-impl<T: Read> AsyncRead for Async<T> {
+/// Types whose I/O trait implementations do not drop the underlying I/O source.
+///
+/// The resource contained inside of the [`Async`] cannot be invalidated. This invalidation can
+/// happen if the inner resource (the [`TcpStream`], [`UnixListener`] or other `T`) is moved out
+/// and dropped before the [`Async`]. Because of this, functions that grant mutable access to
+/// the inner type are unsafe, as there is no way to guarantee that the source won't be dropped
+/// and a dangling pointer won't be left behind.
```

Comment:
I feel the term "pointer" here is a bit odd because I/O safety is about fd.

Reply:
"handle" would make more sense

Reply:
I feel that "pointer" gets across why it's a bad idea to hold the handle after drop in a more serious and direct way. But I'm fine with "handle".